frame/lottery/src/lib.rs

```diff
@@ -447,7 +460,7 @@ impl<T: Config> Pallet<T> {
 				// Create a new ticket.
 				TicketsCount::<T>::put(new_ticket_count);
 				Tickets::<T>::insert(ticket_count, caller.clone());
-				participating_calls.push(call_index);
```

Comment:
not allowed! runtime development 101 says we cannot modify storage, and then return an error.

We must move this line above the `put` and `insert`.

You must make checking this a habit moving forward :)

Reply:
Just putting the `try_push` above the `put` and `insert` is not enough since then I would have to rollback the `transfer` in case of error, right?  
I'm maybe wrong: but I think this case cannot happen anyway since `call_to_index` returns at most `T::MaxCalls` distinct calls and we check that there are no double entries.  
So this could be an `expect`.

Reply:
You can do it before the transfer too.

Reply:
updated